### DISCUSSION TITLE
How to configure clients to work as a remote server

### DISCUSSION THREAD

**Discussion by jin0g** [ltsp/ltsp#233] [Q&A]
Hi, 

I'm trying to manage math calculation servers using LTSP.
It works very well on my Ubuntu Server 18.04.

However, I'm not sure how to **configure client machine to work as a remote server**.

I have two questions:

### Can we set static IPs to clients with ltsp.conf?

I don't want to change the DHCP settings, so I want to associate a static IP with a MAC address, using such as ltsp.conf.

### Is there a way to enable openssh on clients?

Even if 'SSH' is set in KEEP_SYSTEM_SERVICES, the SSH server will not be enabled, because there is no secret key in /etc/ssh/* on client file system.
Even if private key is generated at startup, the secret key will change every reboot.

Please let me know if there is a solution to this problem.
Thank you.

**Comment by enaut** [comment]
Static IP's:
https://github.com/ltsp/community/issues/80#issuecomment-572991699

I don't know about the secret key - I'd guess that it is listed in the excludes.
You could probably do some `echo "secret key" > /etc/ssh/key` in the same section mentioned in the comment above.

**Comment by jin0g** [comment]
@enaut 
I solved my problems-1.
I will try a your solution for second problem.
Thanks a lot.

**Comment by alkisg** [comment]
Try this:

```shell
[server]
OMIT_IMAGE_EXCLUDES="etc/ssh/ssh_host_*"

[clients]
KEEP_SYSTEM_SERVICES="openssh-server"

[mac:address1]
KERNEL_PARAMETERS="ip=192.168.1.101:\${srv}:\${gateway}:\${netmask}"
```

But note that that way you expose the private server keys to your LAN.
It would be better to generate different keys, put them in /etc/ltsp/your-keys, and then move them to /etc/ssh with a POST_INIT_x parameter in ltsp.conf:

```shell
[server]
# This is not needed now: OMIT_IMAGE_EXCLUDES="etc/ssh/ssh_host_*"

[clients]
KEEP_SYSTEM_SERVICES="openssh-server"
POST_INIT_CP_KEYS="mv /etc/ltsp/ssh_host_* /etc/ssh/"

[mac:address1]
KERNEL_PARAMETERS="ip=192.168.1.101:\${srv}:\${gateway}:\${netmask}"

**Comment by jin0g** [comment]
@alkisg Great!
This is a perfect solution for me.
It have worked.
Thank you very much.
